lib/unifex/specs_dsl.ex

```diff
@@ -181,10 +181,14 @@ defmodule Unifex.Specs.DSL do
 
   Elixir definition of %My.Struct{} should contain every field listed in specification and is not generated by Unifex
   """
-  defmacro type(struct) do
+  defmacro type({:"::", _, [_, {:%, _, _}]} = struct) do
```

Comment:
Please update the doc, and also extend the example in the moduledoc so that it contained a struct and an enum